bundles/org.openhab.binding.mqtt.ruuvigateway/src/main/java/org/openhab/binding/mqtt/ruuvigateway/internal/handler/RuuviTagHandler.java

```diff
@@ -0,0 +1,433 @@
+                case CHANNEL_ID_TX_POWER:
+                    atLeastOneRuuviFieldPresent |= updateStateIfLinked(channelUID, ruuvitagData.getTxPower());
+                    break;
+                //
+                // Auxiliary channels, not part of bluetooth advertisement
+                //
+                case CHANNEL_ID_RSSI:
+                    atLeastOneRuuviFieldPresent |= updateStateIfLinked(channelUID, parsed.rssi);
+                    break;
+                case CHANNEL_ID_TS:
+                    atLeastOneRuuviFieldPresent |= updateDateTimeStateIfLinked(channelUID, parsed.ts);
+                    break;
+                case CHANNEL_ID_GWTS:
+                    atLeastOneRuuviFieldPresent |= updateDateTimeStateIfLinked(channelUID, parsed.gwts);
+                    break;
+                case CHANNEL_ID_GWMAC:
+                    atLeastOneRuuviFieldPresent |= updateStringStateIfLinked(channelUID, parsed.gwMac);
+                    break;
+                default:
+                    logger.error("BUG: We have unhandled channel: {}",
+                            thing.getChannels().stream().map(Channel::getUID).collect(Collectors.toList()));
+            }
+        }
+        if (atLeastOneRuuviFieldPresent) {
+            String thingStatusDescription = getThing().getStatusInfo().getDescription();
+            if (getThing().getStatus() != ThingStatus.ONLINE
+                    || (thingStatusDescription != null && !thingStatusDescription.isBlank())) {
+                // Update thing as ONLINE and possibly clear the thing detail status
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            }
+        } else {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Received Ruuvi Tag data but no fields could be parsed: {}", HexUtils.bytesToHex(payload));
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Received Ruuvi Tag data but no fields could be parsed");
+        }
+    }
+
+    @Override
+    public @Nullable ChannelState getChannelState(ChannelUID channelUID) {
+        return channelStateByChannelUID.get(channelUID);
+    }
+
+    @Override
+    protected void updateThingStatus(boolean messageReceived, Optional<Boolean> availabilityTopicsSeen) {
+        // Not used here
+    }
+
+    /**
+     * Update number channel state
+     *
+     * Update is not done when value is null.
+     *
+     * @param channelUID channel UID
+     * @param value value to update
+     * @return whether the value was present
+     */
+    private boolean updateStateIfLinked(ChannelUID channelUID, @Nullable Number value) {
+        RuuviCachedNumberState<?> cache = (RuuviCachedNumberState<?>) channelStateByChannelUID.get(channelUID);
+        if (cache == null) {
+            // Invariant as channels should be initialized already
+            logger.error("Channel {} not initialized. BUG", channelUID);
```

Comment:
As it seems under control/handled, I think warn would suffice here:
```suggestion
            logger.warn("Channel {} not initialized. BUG", channelUID);
```